x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/text_expansion_callout_logic.ts

```diff
@@ -46,19 +48,67 @@ interface TextExpansionCalloutActions {
 }
 
 export interface TextExpansionCalloutValues {
+  createTextExpansionModelError: HttpError | undefined;
   createTextExpansionModelStatus: Status;
   createdTextExpansionModel: CreateTextExpansionModelResponse | undefined;
+  fetchTextExpansionModelError: HttpError | undefined;
   isCreateButtonDisabled: boolean;
   isModelDownloadInProgress: boolean;
   isModelDownloaded: boolean;
   isModelStarted: boolean;
   isPollingTextExpansionModelActive: boolean;
   isStartButtonDisabled: boolean;
+  startTextExpansionModelError: HttpError | undefined;
   startTextExpansionModelStatus: Status;
   textExpansionModel: FetchTextExpansionModelResponse | undefined;
   textExpansionModelPollTimeoutId: null | ReturnType<typeof setTimeout>;
 }
 
+/**
+ * Extracts the topmost error in precedence order (create > start > fetch).
+ * @param createError 
+ * @param fetchError 
+ * @param startError 
+ * @returns the extracted error or null if there is no error
+ */
+export const getTextExpansionError = (
```

Comment:
we can refactor this later, given the current timeline, but this would be better done in a selector. Then the component would just need a single value from `useValues()` instead of 3. We could even call this function from the selector.

Reply:
Agreed - I added an item to the [ELSER tech debt ticket](https://github.com/elastic/enterprise-search-team/issues/4440).